### PR TITLE
Typescript support implemented

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do? 

Implements typescript, which adds readability and reliability.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?

## Testing Checklist

* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by building dist and running.

### CODE REVIEW

File: monkey/monkey_island/cc/ui/package.json [hunk 1]
```diff
@@ -27,85 +27,93 @@
...
+    "react-hot-loader": "^4.13.0",
+    "react-json-tree": "^0.12.1",
     "react-jsonschema-form-bs4": "^1.7.1",
-    "react-particles-js": "^3.3.0",
+    "react-particles-js": "^3.4.1",
     "react-redux": "^5.1.2",
-    "react-router-dom": "^4.3.1",
+    "react-router-dom": "^5.2.0",
     "react-spinners": "^0.9.0",
+    "react-step-wizard": "^5.3.5",
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/package.json:107]: This does not belong here, but will be used shortly in scenarios

File: monkey/monkey_island/cc/ui/package.json [hunk 2]
```diff
@@ -27,85 +27,93 @@
...
-    "@fortawesome/react-fontawesome": "^0.1.11",
+    "@emotion/core": "^10.1.1",
+    "@fortawesome/fontawesome-svg-core": "^1.2.32",
+    "@fortawesome/free-regular-svg-icons": "^5.15.1",
+    "@fortawesome/free-solid-svg-icons": "^5.15.1",
+    "@fortawesome/react-fontawesome": "^0.1.12",
     "@kunukn/react-collapse": "^1.2.7",
-    "bootstrap": "^4.5.2",
+    "@types/jest": "^26.0.15",
+    "@types/node": "^14.14.9",
```
Comment (acepace) [inline monkey/monkey_island/cc/ui/package.json:69]: Shouldn't be a runtime dependency

File: monkey/monkey_island/cc/ui/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "sourceMap": true,
+    "module": "commonjs",
+    "target": "es5",
```
Comment (acepace) [inline monkey/monkey_island/cc/ui/tsconfig.json:5]: Why so low?

File: monkey/monkey_island/cc/ui/package.json [hunk 4]
```diff
@@ -27,85 +27,93 @@
...
+    "react-router-dom": "^5.2.0",
     "react-spinners": "^0.9.0",
+    "react-step-wizard": "^5.3.5",
     "react-table": "^6.10.3",
     "react-toggle": "^4.1.1",
     "react-tooltip-lite": "^1.12.0",
     "redux": "^4.0.4",
     "sha3": "^2.1.3",
-    "snyk": "^1.373.1"
+    "snyk": "^1.427.0",
```
Comment (acepace) [inline monkey/monkey_island/cc/ui/package.json:113]: Shoudln't be a prod dependency

File: monkey/monkey_island/cc/ui/package.json [hunk 5]
```diff
@@ -27,85 +27,93 @@
...
+    "source-map-loader": "^1.1.2",
+    "ts-loader": "^8.0.11",
```
Comment (acepace) [inline monkey/monkey_island/cc/ui/package.json:115]: Should not be a prod dependency

File: monkey/monkey_island/cc/ui/package.json [hunk 6]
```diff
@@ -27,85 +27,93 @@
...
+    "typescript": "^4.1.2"
```
Comment (acepace) [inline monkey/monkey_island/cc/ui/package.json:116]: Should not be a prod dependency

**Review by acepace — CHANGES_REQUESTED** [review]
Note this means that all the current run scripts that just run the existing files will fail if they are TS files.

File: monkey/monkey_island/cc/ui/tsconfig.json (hunk 3 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/tsconfig.json:5]: Upped to es6

File: monkey/monkey_island/cc/ui/package.json (hunk 1 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/package.json:110]: Removed

File: monkey/monkey_island/cc/ui/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "sourceMap": true,
+    "module": "commonjs",
+    "target": "es6",
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/tsconfig.json:5]: image

Not low, this is recommended.

**Review by mssalvatore — APPROVED** [review]
TBH, I have basically no experience with typescript and very limited experience with React and Node. If you addressed @acepace's comments (which it looks to me like you did), then I'm happy with this PR.

**Pull request merged**